src/core/fonts.js

```diff
@@ -1257,8 +1258,20 @@ var Font = (function FontClosure() {
               unicodeCharCode = unicode;
             }
           }
-          this.toFontChar[charCode] = unicodeCharCode;
+          map[+charCode] = unicodeCharCode;
         });
+
+        // Attempt to improve the glyph mapping for (some) composite fonts that
+        // appear to lack meaningful ToUnicode data.
+        if (this.composite && this.toUnicode instanceof IdentityToUnicodeMap) {
+          if (/Verdana/i.test(name)) { // Fixes issue11242_reduced.pdf
+            const GlyphMapForStandardFonts = getGlyphMapForStandardFonts();
+            for (charCode in GlyphMapForStandardFonts) {
+              map[+charCode] = GlyphMapForStandardFonts[charCode];
```

Comment:
I can't help but wonder why we need the plus character here and in other parts of this patch. As far as I can tell from https://github.com/mozilla/pdf.js/blob/de56518799b869d1cb4d313c7628a8aa956966f8/src/core/standard_fonts.js#L269 and other code, character codes should already be numeric. Is that true or am I missing something here?

Reply:
When using the `for (key in object)` format the `key` ends up being converted to a string, which is something you don't want when indexing an Array.

Reply:
Ah, that's it. Makes sense; thanks!